tests/js/keyboardInputTests.js

```diff
@@ -0,0 +1,268 @@
+/*!
+Copyright 2015 OCAD University
+
+Licensed under the New BSD license. You may not use this file except in
+compliance with this License.
+
+You may obtain a copy of the License at
+https://github.com/gpii/universal/LICENSE.txt
+*/
+
+(function ($, fluid) {
+
+    "use strict";
+
+    fluid.registerNamespace("gpii.tests.firstDiscovery.usKeymap");
+    fluid.registerNamespace("gpii.tests.firstDiscovery.keyboardInput");
+
+    gpii.tests.firstDiscovery.charFromKeypressTestCases = [
+        { keyCode: undefined, expected: ""},
+        { keyCode: null, expected: "" },
+        { keyCode: 33, expected: "!" },
+        { keyCode: 48, expected: "0" },
+        { keyCode: 65, expected: "A" },
+        { keyCode: 97, expected: "a" }
```

Comment:
if set, keyCode should always be a number: http://www.w3.org/TR/uievents/#widl-KeyboardEvent-keyCode